### ISSUE TITLE
How to perform Hyperparameter tuning in Yolo-Nas model

### ISSUE THREAD

**Issue by kumarsyamala** [Deci-AI/super-gradients#1304]
### 🚀 Feature Request

I have trained the yolo-Nas model with yolo_m, looking for a method to do hypermeter tuning for yolo_s and yolo_l.

Not only size of the model, are they any other method to choose batch size.

Are they any other parameters to train and if they are what are there and how to train with those parameters?

**Comment by BloodAxe** [comment]
These are base train parameters (common for all recipes by default):

https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/recipes/training_hyperparams/coco2017_yolo_nas_train_params.yaml

And here is Yolo-NAS related parameters (they are derived from base):

https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/recipes/training_hyperparams/default_train_params.yaml

You can override them via command line using hydra override syntax, pass it as dict or create your own yaml file and pass it recipe.

Does this answer your question?

**Comment by kumarsyamala** [comment]
If you dont can you show me the command how to do hyperparameter tuning and, Are there any specific parameters or range of parameters to choose.

And once we train our model with hyperparameter tuning how to know which parameters or good and or not

is there a way to get classification report for it

**Comment by BloodAxe** [comment]
We are currently adding necessary changes to the way we manage our experiments to properly support hyperparameters optimization - https://github.com/Deci-AI/super-gradients/pull/1353
The next step would be to write some docs showing how to use it.

**Comment by kumarsyamala** [comment]
@BloodAxe 

> We are currently adding necessary changes to the way we manage our experiments to properly support hyperparameters optimization - #1353 The next step would be to write some docs showing how to use it.

Any update on how to use hyperparameter tuning on Yolo-Nas model!!?
